### PR TITLE
ship-related render events

### PR DESCRIPTION
**Pull request by alex-s168**
Adds a modifiable ship render event that allows client-side transformation of ship positions and a Shipyard entity render transform event

### CODE REVIEW

File: common/src/main/java/org/valkyrienskies/mod/mixin/mod_compat/vanilla_renderer/MixinLevelRendererVanilla.java [hunk 1]
```diff
@@ -178,12 +179,25 @@ private void redirectRenderChunkLayer(final LevelRenderer receiver,
                 center.x(), center.y(), center.z(),
                 camX, camY, camZ);
 
+            final AtomicReference<Matrix4f> matrix4fNew = new AtomicReference<>(matrix4f);
```
Comment (StewStrong) [inline common/src/main/java/org/valkyrienskies/mod/mixin/mod_compat/vanilla_renderer/MixinLevelRendererVanilla.java:182]: Why are these atomic? Atomic is generally slow so unless this needs to be run on multiple threads we should use something else.

Comment (StewStrong) [reply common/src/main/java/org/valkyrienskies/mod/mixin/mod_compat/vanilla_renderer/MixinLevelRendererVanilla.java:182]: Just write something like `class MutableObject<T> { var object: T }`. Its effectively a modifiable reference but it won't cause lag.

**Comment by Rubydesic** [comment]
Can I see a practical application of this event? It seems like physics mod wanted to update the render transform entirely (which we will be adding support for). We can include this event in the new API (see #608), but there are a lot of parameters here and its not clear which ones are necessary or how this would be used.

**Comment by ewoudje** [comment]
Id like to close this in favor of https://github.com/ValkyrienSkies/Valkyrien-Skies-2/pull/608
This alright for you @alex-s168

**Pull request closed without merging**